src/portable/st/stm32_fsdev/dcd_stm32_fsdev.c

```diff
@@ -1027,64 +1224,74 @@ static bool dcd_write_packet_memory(uint16_t dst, const void *__restrict src, si
   srcVal = src;
   pdwVal = &pma[PMA_STRIDE*(dst>>1)];
 
-  for (i = n; i != 0; i--)
+  while (n--)
   {
-    temp1 = (uint16_t) *srcVal;
+    temp1 = (uint16_t)*srcVal;
     srcVal++;
-    temp2 = temp1 | ((uint16_t)((uint16_t) ((*srcVal) << 8U))) ;
+    temp2 = temp1 | ((uint16_t)(((uint16_t)(*srcVal)) << 8U)) ;
     *pdwVal = temp2;
     pdwVal += PMA_STRIDE;
     srcVal++;
   }
+
+  if (wNBytes & 0x01)
+  {
+    temp1 = *srcVal;
```

Comment:
this would cause an read to out-of-bound for the memory. We should cast it to `uin8_t*` first before doing a read ?

Reply:
No problem I know you are busy:) I'm on vacation I'll look at it later (I think I thought about out of bounds access)

Reply:
Then dont worry about this, I can make changes myself, just want to confirm with you first

Reply:
oh right, sorry, I got confused and mixed-up, all is good then